developer/frameworks/articles/290-configuration/05-reading-configurations-configuration-provider.markdown

```diff
@@ -0,0 +1,115 @@
+# Reading Scoped Configuration Values [](id=reading-configuration-values-from-a-configuration-provider)
+
+If your configuration is scoped to anything other than `SYSTEM`, you have two
+options for reading configuration values.
+
+-   Use `ConfigurationProvider`. This will work for any kind of configuration,
+    and is the only way to read configuration values at the `COMPANY` and
+    `GROUP` scopes.
```

Comment:
We should probably clarify here that GROUP scope is not supported in Site Settings the same way that COMPANY scope is supported in Instance Settings.  For Group scope, ConfigurationProvider still searches the PortletPreferences table.  I'm not sure if that's relevant to the reader of this documentation, but it crossed my mind as something that may cause the user to have certain expectations that may not be met.

Reply:
@drewbrokke I'm guessing this comes out in the wash in differences between the `getCompanyConfiguration()` and `getGroupConfiguration()` methods. Am I on the right track?

If `getGroupConfiguration()` looks for config values from the PortletPreferences table, where does `getCompanyConfiguration()` look?  I just realized I don't actually know what happens to those values after a user enters them in the UI and before `ConfigurationProvider` reads them.

Reply:
@nealrobbins `getGroupConfiguration()` and `getCompanyConfiguration()` look in _both_ the PortletPreferences table and the Configuration table for their data. However, as things currently stand, `getGroupConfiguration()` will never find any group-scoped configurations in the Configurations table.

That being said, I think the only thing we need to mention is that setting a configuration to GROUP scope will _not_ give you an auto-generated UI in Site Settings, but _will_ give you an auto-generated UI in Instance Settings. Setting a configuration to COMPANY scope _will_ give you an auto-generated UI in Instance Settings. All configurations will show up in System Settings unless they are hidden by one of the methods documented elsewhere.

Reply:
@drewbrokke Ok, I got you. I agree that could be more clear, but I want to do it in the introduction, not here.

Overall, we're trying to deal with big picture questions (what, why, whether) in the introductions, and keep tutorials like this one focused on the nuts and bolts of *how*.

Reply:
Got it, sounds good to me. Thanks @nealrobbins.